lib/screens/speaker/speaker.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:ecellapp/core/utils/logger.dart';
+import 'package:ecellapp/models/speaker.dart';
+import 'package:ecellapp/screens/speaker/cubit/speaker_cubit.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class SpeakerScreen extends StatefulWidget {
+  SpeakerScreen({Key key}) : super(key: key);
+
+  @override
+  _SpeakerScreenState createState() => _SpeakerScreenState();
+}
+
+class _SpeakerScreenState extends State<SpeakerScreen> {
+  List<dynamic> speakerResponse;
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: BlocConsumer<SpeakerCubit, SpeakerState>(listener: (context, state) async {
+        if (state is SpeakerSuccess) {
+          await Future.delayed(Duration(seconds: 1));
+        } else if (state is SpeakerError) {
+          Scaffold.of(context).showSnackBar(SnackBar(content: Text(state.message)));
+        }
+      }, builder: (context, state) {
+        if (state is SpeakerSuccess) {
+          speakerResponse = state.json;
+          return _buildSuccess(context);
+        } else if (state is SpeakerLoading) {
+          _speaker();
```

Comment:
since builder is called multiple times depending on how the ui is needed to be built and how flutter builds ui, non-ui functions should be called outside it, either in listener or initState. in this case, since there is a case where the _getSpeakers (change the name) function is to be called again (retry in case of error) it should be called in listener.